presto-spark-base/src/main/java/com/facebook/presto/spark/planner/PrestoSparkRddFactory.java

```diff
@@ -330,49 +324,24 @@ private static Partitioner createPartitioner(PartitioningHandle partitioning, in
         return splits;
     }
 
-    private static List<List<ScheduledSplit>> assignSplitsToTasks(List<ScheduledSplit> splits, int numTasks)
+    private SetMultimap<Integer, ScheduledSplit> assignSplitsToTasks(Session session, PartitioningHandle partitioning, List<ScheduledSplit> splits)
```

Comment:
curious: why using `SetMultimap` instead of `ListMultimap`?

Reply:
TaskSource takes `Set<ScheduledSplit> splits`